lib/ansible/modules/cloud/amazon/iam_role.py

```diff
@@ -496,6 +519,47 @@ def get_attached_policy_list(connection, module, name):
         module.fail_json_aws(e, msg="Unable to list attached policies for role {0}".format(name))
 
 
+def get_role_tags(connection, module):
+    role_name = module.params.get('name')
+    if not hasattr(connection, 'list_role_tags'):
+        return {}
+    try:
+        return boto3_tag_list_to_ansible_dict(connection.list_role_tags(RoleName=role_name)['Tags'])
+    except ClientError:
+        return {}
+
+
+def update_role_tags(connection, module):
+    new_tags = module.params.get('tags')
+    if new_tags is None:
+        return False
+
+    if not hasattr(connection, 'list_role_tags'):
```

Comment:
It might be better to do this check at the beginning of the module, so that in case `list_role_tags` is not available but `module.params['tags'] is not None`, the module fails right away and doesn't start applying *some* changes.

Reply:
(This is not a blocker; especially if this is common for AWS modules, I'm fine with keeping it as-is.)